src/components/Company/PaySchedule/_parts/Edit.tsx

```diff
@@ -0,0 +1,129 @@
+import { Flex, TextField, Select, RadioGroup, NumberField, Grid } from '@/components/Common'
+import { PayPreviewCard } from './PayPreviewCard/PayPreviewCard'
+import { useFormContext } from 'react-hook-form'
+import { ListBoxItem } from 'react-aria-components'
+import { DatePicker } from '@/components/Common/Inputs/DatePicker'
+import { PayScheduleInputs, usePaySchedule } from '../PaySchedule'
+import Spinner from '@/assets/icons/spinner_small.svg?react'
+import { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+
+export const Edit = () => {
+  const { t } = useTranslation('Company.PaySchedule')
+  const { payPeriodPreview, mode, payPreviewLoading } = usePaySchedule()
+  const { control, watch, setValue } = useFormContext<PayScheduleInputs>()
+
+  const frequency = watch('frequency')
+  const custom2xPerMonth = watch('custom_twice_per_month') === 'true' ? true : false
+
+  const firstPayDate = watch('anchor_pay_date')
+  const day1Value = watch('day_1')
+
+  // Keep custom fields in sync with frequency
+  useEffect(() => {
+    if (day1Value !== firstPayDate?.day && firstPayDate?.day && custom2xPerMonth) {
+      setValue('day_1', firstPayDate.day)
+    } else {
+      setValue('day_1', undefined)
+    }
+  }, [custom2xPerMonth, day1Value, firstPayDate, setValue])
+
+  useEffect(() => {
+    if (frequency !== 'Twice per month') {
+      setValue('custom_twice_per_month', undefined)
+    } else if (!custom2xPerMonth) {
+      setValue('day_1', 15)
+      setValue('day_2', 31)
+    }
+  }, [custom2xPerMonth, frequency, setValue])
+
+  if (mode !== 'EDIT_PAY_SCHEDULE' && mode !== 'ADD_PAY_SCHEDULE') {
+    return null
+  }
+
+  return (
+    <div style={{ width: '100%' }}>
+      <Grid gap={4} gridTemplateColumns={{ base: '1fr', small: '1fr 1fr' }}>
+        <div style={{ width: '100%' }}>
```

Comment:
Nit: we could probs set as className instead of inline styles

Reply:
Yes good call out